### PR TITLE
#792 added missing config info in config using default values

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
Now uses config/default_config.json as a reference when missing config keys in order never to have missing info in web interface. Ignores flieds such as exchanges, cryptocurrencies, services and starting portfolio.
This will allow to add more config keys in the future without asking users to update their config.json, we will jsut have to update default_config.json

example with invalid trading/risk
image

### CODE REVIEW

**Comment by Werdor** [comment]
I think you need to add a warning when some keys are missing. The user have to know if some keys are missing and which one.

**Comment by Werdor** [comment]
An easiest and more maintainable way to do this can be the use of a [jsonvalidator](https://github.com/Julian/jsonschema) isn't it ? (with default values)

**Comment by GuillaumeDSM** [comment]
> An easiest and more maintainable way to do this can be the use of a [jsonvalidator](https://github.com/Julian/jsonschema) isn't it ? (with default values)

Good idea, I will add validations

File: config/config_schema.json [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        "files": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "crypto-currencies": {
+      "type": "object"
```
Comment (Herklos) [inline config/config_schema.json]:
```
{
    "type": "object",
    "patternProperties": {
        "^S_": {
            "type": "object",
            "properties": {
                "pairs": {
                    "type": "array",
                    "uniqueItems": true,
                    "items": {
                        "type": "string"
                }
           }
        }
    }
}
```

https://json-schema.org/understanding-json-schema/reference/array.html
https://json-schema.org/understanding-json-schema/reference/object.html

Comment (Herklos) [reply config/config_schema.json]: I didn't test it.

File: config/config_schema.json [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    "data-collector": {
+      "type": "object",
+      "properties": {
+        "symbol": {
+          "type": "string"
+        }
+      }
+    },
+    "exchanges": {
+      "type": "object"
```
Comment (Herklos) [inline config/config_schema.json]:
```
{
    "type": "object",
    "patternProperties": {
        "^S_": { "type": "object" }
    }
}
```

File: config/config_schema.json [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      },
+      "required": [
+        "global-info",
+        "price-alerts",
+        "trades",
+        "notification-type"
+      ]
+    },
+    "services": {
+      "type": "object"
```
Comment (Herklos) [inline config/config_schema.json]:
```
{
    "type": "object",
    "patternProperties": {
        "^S_": { "type": "object" }
    }
}
```

File: config/config_schema.json [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+              "maximum": 100
+            }
+          },
+          "required": [
+            "maker",
+            "taker"
+          ]
+        },
+        "starting-portfolio": {
+          "type": "object"
```
Comment (Herklos) [inline config/config_schema.json]:
```
{
    "type": "object",
    "patternProperties": {
        "^S_": { "type": "number" }
    }
}
```

File: config/config_schema.json [hunk 5]
```diff
@@ -0,0 +1,171 @@
+{
+  "type": "object",
+
+  "properties": {
+    "DEBUG": {
+      "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": false

File: config/config_schema.json [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    },
+    "DEV-MODE": {
+      "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": false

File: config/config_schema.json [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    },
+    "PERF": {
+      "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": false

File: config/config_schema.json [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+    },
+    "SAVE_EVALUATIONS": {
+      "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": false

File: config/config_schema.json [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    },
+    "performance-analyser": {
+      "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": false

File: config/config_schema.json [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+    },
+    "backtesting": {
+      "type": "object",
```
Comment (Herklos) [inline config/config_schema.json]: "default": {},

File: config/config_schema.json (hunk 1 above)
Comment (Herklos) [inline config/config_schema.json]:
```
"default": {
   "Bitcoin": {
      "pairs" : ["BTC/USDT"]
    }
}
```

File: config/config_schema.json [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+            "type": "string"
+          }
+        }
+      }
+    },
+    "crypto-currencies": {
+      "type": "object"
+    },
+    "data-collector": {
+      "type": "object",
```
Comment (Herklos) [inline config/config_schema.json]: "default": {},

File: config/config_schema.json (hunk 2 above)
Comment (Herklos) [inline config/config_schema.json]:
```
"default": {
    "binance": {
      "api-key": "your-api-key-here",
      "api-secret": "your-api-secret-here"
    }
  }
```

File: config/config_schema.json [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+        "symbol": {
+          "type": "string"
+        }
+      }
+    },
+    "exchanges": {
+      "type": "object"
+    },
+    "metrics": {
+      "type": "object",
```
Comment (Herklos) [inline config/config_schema.json]: "default": {},

File: config/config_schema.json [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+        "enabled": {
+          "type": "boolean"
+        },
+        "metrics-bot-id": {
+          "type": "string"
+        }
+      }
+    },
+    "notification": {
+      "type": "object",
```
Comment (Herklos) [inline config/config_schema.json]: "default": {},

File: config/config_schema.json [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+        "price-alerts",
+        "trades",
+        "notification-type"
+      ]
+    },
+    "services": {
+      "type": "object"
+    },
+    "tentacles-packages": {
+      "type": "array",
```
Comment (Herklos) [inline config/config_schema.json]: "default": {},

File: config/config_schema.json [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+      "items": {
+        "type": "string"
+      }
+    },
+    "trader": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": false

File: config/config_schema.json [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "trader-simulator": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": true

File: config/config_schema.json [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+        },
+        "fees": {
+          "type": "object",
+          "properties": {
+            "maker": {
+              "type": "number",
+              "minimum": 0,
+              "maximum": 100
```
Comment (Herklos) [inline config/config_schema.json]: "default": 0.1

File: config/config_schema.json [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+            },
+            "taker": {
+              "type": "number",
+              "minimum": 0,
+              "maximum": 100
```
Comment (Herklos) [inline config/config_schema.json]: "default": 0.1

File: config/config_schema.json (hunk 4 above)
Comment (Herklos) [inline config/config_schema.json]:
```
"default": {
      "BTC": 10,
      "USDT": 1000
}
```

File: config/config_schema.json [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+      "type": "object",
+      "properties": {
+        "global-info": {
+          "type": "boolean"
+        },
+        "price-alerts": {
+          "type": "boolean"
+        },
+        "trades": {
+          "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": true

File: config/config_schema.json [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+      }
+    },
+    "notification": {
+      "type": "object",
+      "properties": {
+        "global-info": {
+          "type": "boolean"
+        },
+        "price-alerts": {
+          "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": true

File: config/config_schema.json [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+        "metrics-bot-id": {
+          "type": "string"
+        }
+      }
+    },
+    "notification": {
+      "type": "object",
+      "properties": {
+        "global-info": {
+          "type": "boolean"
```
Comment (Herklos) [inline config/config_schema.json]: "default": true

**Pull request merged**